clients/vscode-hlasmplugin/scripts/syntaxes/hlasm_tmgrammar_template.txt

```diff
@@ -386,24 +412,51 @@
       "name": "constant.numeric.hlasm"
     },
     "variable": {
-      "match": "&(([A-Za-z@#$_][A-Za-z@#$_0-9]*)|([(].*?[)]))(?<=^${listingDetails}$.{0,71})",
-      "name": "variable.other.readwrite.hlasm"
+      "match": "(?:(?:&[A-Za-z@#$_][A-Za-z@#$_0-9]*)|(&[(].*?[)])([(].*?[)])?)(?<=^${listingDetails}$.{0,71})",
+      "name": "variable.other.readwrite.hlasm",
+      "captures": {
+        "1": {
+          "patterns": [
+            {
+              "include": "#variableSpecialChars"
+            }
+          ]
+        },
+        "2": {
+          "patterns": [
+            {
+              "include": "#variableSpecialChars"
+            },
+            {
+              "include": "#asmArguments"
+            }
+          ]
+        }
+      }
     },
-    "ordinarySymbol": {
-      "match": "[A-Za-z@#$_][A-Za-z@#$_0-9]*(?<=^${listingDetails}$.{0,71})",
-      "name": "support.class.hlasm"
+    "variableSpecialChars": {
+      "match": "[().]",
+      "name": "keyword.operator.hlasm"
     },
     "asmSpecialStatements": {
       "match": "(?<=[=(),*+\\s])=?\\d*(A|B|C|D|E|F|G|H|P|V|X|Z)(L\\d*)?((?=[=(),*\\s+]|$)(?<=^${listingDetails}$.{0,71}))",
       "name": "punctuation.definition.tag.hlasm"
     },
     "continuationSymbol": {
-      "match": "\\s*(?<=^${listingDetails}$.{71})\\S",
-      "name": "keyword.other.hlasm"
+      "match": "\\s*(?<=^${listingDetails}$.{71})(\\S)",
+      "captures": {
+        "1": {
+          "name": "keyword.other.hlasm"
+        }
+      }
     },
     "sequenceNumbers": {
-      "match": "\\s*.?(?<=^${listingDetails}$.{72}).+$",
-      "name": "invalid.ignored"
+      "match": "\\s*.?(?<=^${listingDetails}$.{72})\\s*(\\S(?:.*\\S)?)\\s*?$",
```

Comment:
Everything after 72 is ignored (and corresponding stuff in listings) whether it is a space or not so I think just `(.+)` would be appropriate.